packages/components/src/core/Autocomplete/components/AutocompleteMultiColumn/index.tsx

```diff
@@ -219,6 +243,21 @@ const AutocompleteMultiColumn = <
              * mobile keyboard
              */
             inputMode: "text",
+            /**
+             * (masoudmanson): This is to handle the Escape key to close the dropdown.
+             * If the dropdown is controlled, we call the onClick prop to allow the parent
+             * to handle the closing of the dropdown.
+             */
+            onKeyDown: (event: React.KeyboardEvent) => {
+              if (event.key === "Escape") {
+                if (isOpenControlled) {
+                  onClick?.();
+                  return;
+                }
+
+                setPopperOpen(false);
```

Comment:
Oh I wonder if we should make it consistent with other event handlers in the sense that we call `props.onKeyDown()` instead of `props.onClick?.()`, so if the component is controlled, the parent will need to handle the onKeyDown event on their own and decide if they want to set open to false and pass it down to the component?

Reply:
Oh, of course, that makes sense, I'll fix it. :D

Reply:
Thank you!!